src/duckdb.ts

```diff
@@ -0,0 +1,97 @@
+import {existsSync} from "node:fs";
+import {mkdir, writeFile} from "node:fs/promises";
+import {dirname, join} from "node:path/posix";
+import {cross} from "d3-array";
+import type {DuckDBConfig} from "./config.js";
+import {faint} from "./tty.js";
+
+const downloadRequests = new Map<string, Promise<string>>();
+
+export async function duckDBManifest(
+  duckdb: DuckDBConfig,
+  {root, log, aliases}: {root: string; log?: boolean; aliases?: Map<string, string>}
+) {
+  return {
+    bundles: duckdb.bundles,
+    extensions: await Promise.all(
+      cross(duckdb.bundles, duckdb.install).map(async ([p, name]) => {
+        let ext = await resolveDuckDBExtension(root, p, duckdb.source[name], name);
+        if (aliases?.has(ext)) ext = aliases.get(ext)!;
+        return [
+          name,
+          {
+            ref: dirname(dirname(dirname(ext))),
+            load: duckdb.load.includes(name),
+            bundle: p
+          }
+        ];
+      })
+    ),
+    log
+  };
+}
+
+/**
+ * Given a duckdb configuration and an extension name such as "parquet", saves
+ * the binary to _duckdb/{hash}/v1.1.1/wasm_{p}/parquet.duckdb_extension.wasm
+ * for every supported platform p ("eh" and "mvp"), and returns a content-hashed
+ * reference (_duckdb/{hash}) to use in the corresponding DuckDB INSTALL
+ * statement. The repo is structured as required by DuckDB with:
+ * ${repo}/v1.1.1/wasm_{p}/${name}.duckdb_extension.wasm
+ */
+export async function resolveDuckDBExtension(root: string, p: string, repo: string, name: string): Promise<string> {
```

Comment:
The `p` name here isn’t great. I try to only use cryptic single-letter variables for short-lived local variables; function arguments should ideally have more descriptive names, especially for functions that are exported from the module. Should be `platform` instead, right?

Reply:
I think so, though they use the word "bundle", and adjust it to the "platform" (which are characteristics of the browser) in the `selectBundle` function:
https://github.com/duckdb/duckdb-wasm/blob/a9a38bb480c1b44bb90e6fa35e731c572e8868ef/packages/duckdb-wasm/src/platform.ts#L27